utilities/cache_utils.py

```
@@ -0,0 +1,469 @@
+    cache_dir = Path("cache/prefixes")
+    cache_dir.mkdir(parents=True, exist_ok=True)
+    return cache_dir
+
+def save_to_disk(cache_key:str, cache_type:str, speaker_embedding: torch.Tensor):
+    try:
+        if cache_type == "prefix":
+            cache_dir = get_prefix_cache_dir()
+        elif cache_type == "embeds":
+            cache_dir = get_embed_cache_dir()
+        else:
+            raise ValueError(f"Unknown cache type: {cache_type}")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)